js/packages/mobile-wallet-adapter-protocol/src/transact.ts

```diff
@@ -175,7 +193,29 @@ export async function transact<TReturn>(
                                         ),
                                     );
                                     return new Promise((resolve, reject) => {
-                                        jsonRpcResponsePromises[id] = { resolve, reject };
+                                        jsonRpcResponsePromises[id] = {
+                                            resolve(result) {
+                                                switch (p) {
+                                                    case 'authorize':
+                                                    case 'reauthorize': {
+                                                        const { wallet_uri_base } = result as Awaited<
+                                                            ReturnType<MobileWallet['authorize' | 'reauthorize']>
+                                                        >;
+                                                        if (wallet_uri_base != null) {
+                                                            try {
+                                                                assertSecureEndpointSpecificURI('http://apple.com');
```

Comment:
Cool, but test code, I assume?

Reply:
No. If that throws, we need to catch it and send it through `reject()` so that the promise for `wallet.authorize()` rejects with the thrown error. Otherwise the whole system halts because the `authorize` promise never resolves _nor_ rejects.

Reply:
OH. Haha.

* scrolls over *

Notices the URL.